### PR TITLE
Improvements

### PR DESCRIPTION
**Pull request by markuskramerIgitt**
Notes:
 
- Implemented  MINION_ID_FUNCTION, MINION_CONFIG, CONFIG_TYPE
- CONFIG_TYPE silent installer only
- Refactored the msi installer (WiX and the C# code )
- Added documentation (website and code)
- Changed program code to follow naming convention (_DECAC and _CADH)
- Using Visual C++ runtime redistributables from repo.saltstack.com
- Added  open_last_log_in_code.cmd

### CODE REVIEW

**Comment by xeacott** [comment]
Oh! Is this to allow adding a custom minion id?

**Comment by markuskramerIgitt** [comment]
Added comment

**Comment by xeacott** [comment]
Would you be able to write in the description of the PR a change log that states all of the changes being made? I would find it very useful

**Comment by markuskramerIgitt** [comment]
> Would you be able to write in the description of the PR a change log that states all of the changes being made? I would find it very useful

Just done

File: wix.d/MinionConfigurationExtension/MinionConfiguration.cs [hunk 1]
```diff
@@ -515,141 +664,67 @@ private static ActionResult replace_pattern_in_one_config_file
...
+				session.Log("...Move_dir!   " + ffn);
+				if (Directory.Exists(target)) {
+					session.Log("...target exists   " + target);
+				} else {
+					Directory.Move(ffn, ffn + timestamp_bak);
+				}
+			}
+		}
+	}
 }
```
Comment (twangboy) [inline wix.d/MinionConfigurationExtension/MinionConfiguration.cs]: This entire file has a mixture of tabs and spaces... we should use one or the other. Preferably spaces....

Comment (markuskramerIgitt) [reply wix.d/MinionConfigurationExtension/MinionConfiguration.cs]: I did not notice and agree. 
I'll be back next week.

Comment (markuskramerIgitt) [reply wix.d/MinionConfigurationExtension/MinionConfiguration.cs]: See https://github.com/markuskramerIgitt/salt-windows-msi/commit/1bd264644e6bad0c8120cd2cc4648d6af7933723

**Comment by twangboy** [comment]
@markuskramerIgitt Is this ready?

**Comment by xeacott** [comment]
This is a lot of changes. I briefly skimmed them and realized quickly; but these are all good. I'll need to digest these changes, but I trust they're needed and improve things so I will also approve.

**Comment by markuskramerIgitt** [comment]
There is an important commit (in my fork), which does not appear here.

https://github.com/markuskramerIgitt/salt-windows-msi/commit/3d54917148feba43ee9c0f8a7b3df5f7c9c5ea97

**Pull request merged**